### PR TITLE
Remove a bunch of unused test code

### PR DESCRIPTION
**Pull request by Eilon**
The tests were testing nothing but themselves!

@Tratcher @HaoK

### CODE REVIEW

File: test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/OpenIdConnectHandlerTests.cs [hunk 1]
```diff
@@ -30,29 +29,6 @@ public class OpenIdConnectHandlerTests
         private static SecurityToken specCompliantOpenIdConnect = new JwtSecurityToken("issuer", "audience", new List<Claim> { new Claim("iat", EpochTime.GetIntDate(DateTime.UtcNow).ToString()), new Claim("nonce", nonceForOpenIdConnect) }, DateTime.UtcNow, DateTime.UtcNow + TimeSpan.FromDays(1));
         private const string ExpectedStateParameter = "expectedState";
 
-        /// <summary>
-        /// Sanity check that logging is filtering, hi / low water marks are checked
-        /// </summary>
-        [Fact]
-        public void LoggingLevel()
```
Comment (Eilon) [inline test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/OpenIdConnectHandlerTests.cs]: This test was testing test-only code.

File: test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/OpenIdConnectMiddlewareForTestingAuthenticate.cs [hunk 2]
```diff
@@ -34,9 +34,6 @@ public class OpenIdConnectMiddlewareForTestingAuthenticate : OpenIdConnectMiddle
         : base(next, dataProtectionProvider, loggerFactory, encoder, services, sharedOptions, options, htmlEncoder)
         {
             _handler = handler;
-            var customFactory = loggerFactory as InMemoryLoggerFactory;
-            if (customFactory != null)
-                Logger = customFactory.Logger;
```
Comment (Eilon) [inline test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/OpenIdConnectMiddlewareForTestingAuthenticate.cs]: Nothing is actually using this (all the tests pass).

File: test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/TestUtilities.cs [hunk 3]
```diff
@@ -14,97 +13,7 @@ public class TestUtilities
     {
         public const string DefaultHost = @"http://localhost";
 
-        public static bool AreEqual<T>(object obj1, object obj2, Func<object, object, bool> comparer = null) where T : class
```
Comment (Eilon) [inline test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/TestUtilities.cs]: None of these used at all.

File: test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/LoggingUtilities.cs [hunk 4]
```diff
@@ -1,148 +0,0 @@
...
-// they can be reviewed for general content.
-
-using System;
-using System.Collections.Generic;
-using System.Text;
-using Microsoft.Extensions.Logging;
-
-namespace Microsoft.AspNet.Authentication.Tests.OpenIdConnect
-{
-    public class LoggingUtilities
```
Comment (Eilon) [inline test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/LoggingUtilities.cs]: Totally not used.

**Comment by HaoK** [comment]
Does that make it possible to get > 100% code coverage then?

**Comment by HaoK** [comment]
Yay :shipit:

**Comment by Eilon** [comment]
Not sure about >%100 but we'll see :smile:

**Comment by Eilon** [comment]
Thanks!

**Pull request merged**